tests/foreman/cli/test_model.py

```diff
@@ -58,88 +75,54 @@ def test_positive_create_with_vendor_class(self):
         """Don't create an Model with invalid data.
 
         :id: b2eade66-b612-47e7-bfcc-6e363023f498
 
+        :parametrized: yes
+
         :expectedresults: Model is not created.
 
         :CaseImportance: High
         """
-        for name in invalid_values_list():
-            with self.subTest(name):
-                with self.assertRaises(CLIReturnCodeError):
-                    Model.create({'name': name})
-
-    @tier1
-    def test_positive_update_name(self):
-        """Successfully update an Model.
-
-        :id: 66eb6cf2-9ec5-4947-97e0-b612780c5cc3
-
-        :expectedresults: Model is updated.
-
-        :CaseImportance: Medium
-        """
-        model = make_model()
-        for new_name in valid_data_list().values():
-            with self.subTest(new_name):
-                Model.update({'id': model['id'], 'new-name': new_name})
-                model = Model.info({'id': model['id']})
-                self.assertEqual(model['name'], new_name)
+        with pytest.raises(CLIReturnCodeError):
+            Model.create({'name': name})
 
     @tier1
-    def test_negative_update_name(self):
+    @pytest.mark.parametrize('new_name', **parametrized(invalid_values_list()))
+    def test_negative_update_name(self, class_model, new_name):
         """Create Model then fail to update its name
 
         :id: 98020a4a-1789-4df3-929c-6c132b57f5a1
 
+        :parametrized: yes
+
         :expectedresults: Model name is not updated
 
         :CaseImportance: Medium
         """
-        model = make_model()
-        for new_name in invalid_values_list():
-            with self.subTest(new_name):
-                with self.assertRaises(CLIReturnCodeError):
-                    Model.update({'id': model['id'], 'new-name': new_name})
-                result = Model.info({'id': model['id']})
-                self.assertEqual(model['name'], result['name'])
-
-    @tier1
-    @upgrade
-    def test_positive_delete_by_id(self):
-        """Create Model with valid values then delete it
-        by ID
-
-        :id: 39f02cec-ac4c-4801-9a4a-11160247213f
-
-        :expectedresults: Model is deleted
-
-        :CaseImportance: High
-        """
-        for name in valid_data_list().values():
-            with self.subTest(name):
-                model = make_model({'name': name})
-                Model.delete({'id': model['id']})
-                with self.assertRaises(CLIReturnCodeError):
-                    Model.info({'id': model['id']})
+        with pytest.raises(CLIReturnCodeError):
+            Model.update({'id': class_model['id'], 'new-name': new_name})
+        result = Model.info({'id': class_model['id']})
+        assert class_model['name'] == result['name']
 
     @tier1
-    def test_negative_delete_by_id(self):
+    @pytest.mark.parametrize('entity_id', **parametrized(invalid_id_list()))
+    def test_negative_delete_by_id(self, entity_id):
         """Create Model then delete it by wrong ID
```

Comment:
Just pointing out that the comment should probably be updated, since the test doesn't create the Model.